src/Analysis/GWASWizard/wizardEndpoints/cohortMiddlewareApi.js

```diff
@@ -60,30 +60,86 @@ export const fetchOverlapInfo = async (
   return getOverlapStats.json();
 };
 
-export const fetchConceptStatsByHareForCaseControl = async (
-  queriedCohortDefinitionId,
-  otherCohortDefinitionId,
-  selectedCovariates,
-  selectedDichotomousCovariates,
+export const filterSubsetCovariates = (subsetCovariates) => {
+  const filteredSubsets = [];
+  subsetCovariates.forEach((covariate) => {
+    if (covariate.variable_type === 'custom_dichotomous') {
+      filteredSubsets.push({
+        cohort_ids: covariate.cohort_ids,
+        provided_name: covariate.provided_name,
+        variable_type: covariate.variable_type,
+      });
+    } else {
+      filteredSubsets.push({
+        variable_type: 'concept',
+        concept_id: covariate.concept_id,
+      });
+    }
+  });
+  return filteredSubsets;
+};
+
+export const fetchConceptStatsByHareSubset = async (
+  cohortDefinitionId,
+  subsetCovariates,
   sourceId,
 ) => {
+  const variablesPayload = {
+    variables:
+      [
+        ...filterSubsetCovariates(subsetCovariates),
+      ],
+  };
+  const conceptStatsEndPoint = `${cohortMiddlewarePath}concept-stats/by-source-id/${sourceId}/by-cohort-definition-id/${cohortDefinitionId}/breakdown-by-concept-id/${hareConceptId}`;
+  const reqBody = {
+    method: 'POST',
+    credentials: 'include',
+    headers,
+    body: JSON.stringify(variablesPayload),
+  };
+  const getConceptStats = await fetch(conceptStatsEndPoint, reqBody);
+  return getConceptStats.json();
+};
+
+export const addCDFilter = (cohortId, otherCohortId, covariateArr) => {
```

Comment:
please update the comment below to reflect the new variable names in this function

Reply:
ty, edited